compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureDesugar.java

```diff
@@ -324,7 +378,9 @@ private void createFunctionMap(BLangFunction funcNode, SymbolEnv funcEnv) {
         if (funcNode.body == null) {
             funcNode.body = ASTBuilderUtil.createBlockFunctionBody(funcNode.pos);
         }
-        ((BLangBlockFunctionBody) funcNode.body).stmts.add(0, mapVarDef);
+        if (funcNode.body.getKind() == NodeKind.BLOCK_FUNCTION_BODY) {
```

Comment:
Why do we need this check now?

Reply:
Because external functions haven't function bodies.

Reply:
This was always the case. My question was more about why we needed to introduce this change now?